scheduler/core/components/optimizer/greedymax.py

```diff
@@ -374,6 +374,26 @@ def _charge_time(observation: Observation, atom_start: int = 0, atom_end: int =
             observation.sequence[n_atom].observed = True
             observation.sequence[n_atom].qa_state = QAState.PASS
 
+    def _update_score(self, program) -> None:
+        """Update the scores of the incomplete groups in the scheduled program"""
+
+        print("Call score_program")
+        program_calculations = self.selection.score_program(program)
```

Comment:
Because of the initial inferred type of `self.selection` as I mentioned above, this generates a warning. When we change the constructor `self.simulation` line, that warning goes away.